src/data/files.rs

```diff
@@ -1,19 +1,41 @@
 use crate::prelude::*;
 use nu_errors::ShellError;
 use nu_protocol::{TaggedDictBuilder, UntaggedValue, Value};
+use std::path::PathBuf;
 
 pub(crate) fn dir_entry_dict(
-    filename: &std::path::Path,
-    metadata: &std::fs::Metadata,
+    path: &PathBuf,
     tag: impl Into<Tag>,
     full: bool,
+    short_name: bool,
 ) -> Result<Value, ShellError> {
-    let mut dict = TaggedDictBuilder::new(tag);
-    dict.insert_untagged("name", UntaggedValue::string(filename.to_string_lossy()));
+    let tag = tag.into();
+    let metadata = match std::fs::metadata(path) {
```

Comment:
I think we should keep the `metadata` parameter.

First, we can no longer inject a dependency (`Metadata`), which makes it hard to test and work with this function.

Second, you've removed the control of what kind of metadata we're getting. By switching to `metadata`, we can no longer do `symlink_metadata`, meaning we _always_ traverse the symlink. I think this function should focus on formatting and creating a `Value`, with very little opinion on how to get all the data to do so. As a side note, the fact that no tests failed tells me we're missing one 🙂

Reply:
> First, we can no longer inject a dependency (`Metadata`), which makes it hard to test and work with this function.
I don't understand what this means, exactly.

>  you've removed the control of what kind of metadata we're getting. By switching to `metadata`, we can no longer do `symlink_metadata`, meaning we always traverse the symlink.
Oh, I suppose I forgot that this isn't a local helper function.

Reply:
> I don't understand what this means, exactly.

No worries. I'm talking about [dependency injection](https://en.wikipedia.org/wiki/Dependency_injection). Essentially, you pass instances of a dependency (in Rust, that'd be a struct) into functions as much as possible, instead of constructing them inside the functions.

> Oh, I suppose I forgot that this isn't a local helper function.

Unfortunately, it isn't :(

My recommendation would be to revert this part of of your PR, and allow passing in `metadata: &std::fs::Metadata`. Will that break something you're trying to do?